indexing-hadoop/src/main/java/org/apache/druid/indexer/JobHelper.java

```diff
@@ -482,20 +475,16 @@ public long push() throws IOException
         .withSize(size.get())
         .withBinaryVersion(SegmentUtils.getVersionFromDir(mergedBase));
 
-    if (!renameIndexFiles(outputFS, tmpPath, finalIndexZipFilePath)) {
-      throw new IOE(
-          "Unable to rename [%s] to [%s]",
-          tmpPath.toUri().toString(),
-          finalIndexZipFilePath.toUri().toString()
-      );
-    }
-
-    return finalSegment;
+    return new DataSegmentAndIndexZipFilePath(
+        finalSegment,
+        tmpPath.toUri().getPath(),
+        finalIndexZipFilePath.toUri().getPath()
+    );
   }
 
   public static void writeSegmentDescriptor(
```

Comment:
this method also deletes and creates a file, I think the descriptor creation should also be moved into the main task (it could be handled in `renameIndexFile` where you have access to a FileSystem)

Reply:
We seem to be storing the information about the segments that we publish in the descriptor file, and then read the data written to this this file / directory in main task in order to know the list of segments that were produced. If we dont create this file in the sub task / job, how will we know what segments were created?

Reply:
Changed it so that the segment descriptor for a segment is not overwritten if it already exists, as we discussed.